wavedump/src/wavedump.c

```diff
@@ -2114,26 +2287,30 @@ int main(int argc, char *argv[])
             }
             nread += 1;
         }
+	
+	// print_wfdata(wfdata);
 
         if (nread > 0) {
             printf("writing %i events to file\n", nread);
-            if (add_to_output_file(output_filename, wfdata, nread, chmask, nsamples, &WDcfg)) {
+            if (add_to_output_file(output_filename, wfdata, bdata, nread, chmask, nsamples, &WDcfg)) {
                 fprintf(stderr, "failed to write events to file! quitting...\n");
                 exit(1);
             }
         }
 
         total_events += nread;
-
-        usleep(100000);
+	
+	// printf("before\n");
+        usleep(1000);
+	// printf("after\n");
     }
 
     if (stop)
         fprintf(stderr, "ctrl-c caught. writing out %i events\n", nread);
 
-    if (nread > 0 && add_to_output_file(output_filename, wfdata, nread, chmask, nsamples, &WDcfg)) {
-        fprintf(stderr, "failed to write events to file!\n");
-    }
+    // if (nread > 0 && add_to_output_file(output_filename, wfdata, nread, chmask, nsamples, &WDcfg)) {
+    //     fprintf(stderr, "failed to write events to file!\n");
+    // }

```

Comment:
Why is this deleted?

Reply:
Let me know if I got this wrong, but I think we're already preforming this check inside of the acquisition loop (just a few lines up). If we do the check twice, then the data is written twice to the output file since the if statement calls `add_to_output_file`.

Reply:
ah ok I think I see the problem. I think in the block where we rite to the output file in the loop we should set nread = 0 afterwards. This line is then supposed to write out any events which had been saved but we hit ctrl-c to stop it.